pkg/client/client.go

```diff
@@ -25,6 +25,7 @@ import (
 // TLSBootstrapClient retrieves tokens for performing node TLS bootstrapping.
 type TLSBootstrapClient interface {
 	GetBootstrapToken(ctx context.Context) (string, error)
+	setupClientConnection(ctx context.Context) (*grpc.ClientConn, error)
```

Comment:
does this absolutely need to be part of the interface?

now that I think about it I'm wondering if we even need this interface at all...

Reply:
I think we can probably get rid of this interface altogether, maybe something for a separate PR...

Reply:
I think we can revisit this interface altogether in a separate PR, though for now `setupClientConnection` shouldn't need to be a part of it.